### PR TITLE
feat(api): implement finding planning area for project

### PR DESCRIPTION
**Pull request by Dyostiq**
## Substitute this line for a meaningful title for your change

I've decided to move all planning-area-related things to a separate module. Otherwise, it was hard for me to inter actual usages and consequences.

### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/api/src/modules/projects/planning-areas/admin-planning-areas.repository.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export class AdminPlanningAreasRepository {
+  constructor(private readonly adminAreasService: AdminAreasService) {}
+
+  async findAdminAreaIdByLevelId(adminAreaId: string) {
+    const foundId:
+      | Pick<AdminArea, 'id'>
+      | NotFound = await this.adminAreasService
+      .getByLevel1OrLevel2Id(adminAreaId, {
+        fields: ['id'],
+      })
```
Comment (kgajowy) [inline api/apps/api/src/modules/projects/planning-areas/admin-planning-areas.repository.ts:17]: shouldn't it catch there with some other type than `not found` ?

**Review by kgajowy — APPROVED** [review]

File: api/apps/api/src/modules/projects/planning-areas/admin-planning-areas.repository.ts (hunk 1 above)
Comment (Dyostiq) [reply api/apps/api/src/modules/projects/planning-areas/admin-planning-areas.repository.ts:17]: I am not sure, I do not expect any other error here so I'm not even sure how should we react to it

Comment (hotzevzl) [reply api/apps/api/src/modules/projects/planning-areas/admin-planning-areas.repository.ts:17]: we explicitly only throw NotFoundException, then of course stuff can happen in TypeORM or whatnot, but I'd say this is ok as is.

**Review by hotzevzl — APPROVED** [review]
@Dyostiq thank you, and apologies for having taken so long to set aside the time and focus needed to review this: after all it's much more readable than what one could fear by looking at "37 Files changed in one commit" 😉 thanks to the neat setup of abstractions you've added here.

Please merge - I suppose you will want to update the error handling parts to use io-ts in a separate chore PR?

**Comment by Dyostiq** [comment]
@hotzevzl Many thanks for the review :) Yes, I'll upgrade it to fp-ts in another PR.

**Pull request merged**